### PR TITLE
ctrruntimeconfig: Resync on failure and accuratey detect object updates

### PR DESCRIPTION
**Pull request by umohnani8**
When a CR was being created and the label didn't exist in any of the pools
the controller would fail and exit it out. It should try resyncing so that
when the label is added later on, it can finally sync properly and make
the necessary changes to the config fies.
Also adds patch to only update when a real change in the containerruntimeconfig
is detected. It also fetches a fresh config so that we limit the number of "object
has been updated" messages.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1686282

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@runcom @mrunalp @rphillips PTAL

**Comment by kikisdeliveryservice** [comment]
Does this need updated tests? Seems like 2 funcs were changed a lot.

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go [hunk 1]
```diff
@@ -366,12 +379,26 @@ func (ctrl *Controller) generateOriginalContainerRuntimeConfigs(role string) (*i
...
+		return lerr
+	})
+	// If an error occured in updating the status just log it
+	if statusUpdateErr != nil {
+		glog.Warningf("error updating container runtime config status: %v", statusUpdateErr)
 	}
-	_, lerr := ctrl.client.MachineconfigurationV1().ContainerRuntimeConfigs().UpdateStatus(cfg)
-	return lerr
+	// Want to return the actual error received from the sync function
+	return err
```
Comment (kikisdeliveryservice) [inline pkg/controller/container-runtime-config/container_runtime_config_controller.go]: I think the `lerr` `err` is kind of confusing. Perhaps renaming the func argument `err` to something like syncErr and renaming `lerr` -> `err` would be a little more clear?

Comment (umohnani8) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go]: Renamed it to `updateErr`, hopefully that is more clear.

**Comment by umohnani8** [comment]
@kikisdeliveryservice the tests should be fine. This is just ensuring that the correct error is being returned for the sync functions.

**Comment by umohnani8** [comment]
Tests are green. @cgwalters @runcom PTAL

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go (hunk 1 above)
Comment (kikisdeliveryservice) [reply pkg/controller/container-runtime-config/container_runtime_config_controller.go]: thanks this is perfect!

**Pull request merged**